Kitodo/src/main/java/org/kitodo/services/data/UserService.java

```diff
@@ -184,6 +203,44 @@ private void manageUserGroupsDependenciesForIndex(User user) throws CustomRespon
         }
     }
 
+    /**
+     * Gets user by login.
+     *
+     * @param login
+     *      The login.
+     * @return
+     *      The user.
+     */
+    public User getByLogin(String login) throws DAOException {
+        List<User> users = getByQuery("from User where login = :username", "username", login);
+        if (users.size() == 1)  {
+            return users.get(0);
+        } else if (users.size() == 0) {
+            throw new UsernameNotFoundException("Username " + login + "not found!");
+        } else {
+            throw new UsernameNotFoundException("Username " + login + "was found more than once");
+        }
+    }
+
+    /**
+     * Gets the current authenticated user.
+     *
+     * @return
+     *      The user.
+     */
+    public User getAuthenticatedUser() throws DAOException {
+        SecurityContext securityContext = SecurityContextHolder.getContext();
+        Authentication authentication = securityContext.getAuthentication();
+        if (authentication != null) {
+            Object principal = authentication.getPrincipal();
+            UserDetails userDetails = principal instanceof UserDetails ? (UserDetails) principal : null;
+            if (userDetails != null) {
+                return getByLogin(userDetails.getUsername());
+            }
+        }
+        return null;
```

Comment:
Is returning `null` a good solution for this use case?

Reply:
See my answer to your comment of LoginForm

Reply:
I'm not happy about `null` return values but needed changes are complex and not only related to this changes in this pull request.